### PR TITLE
Fixed block explorer links

### PR DESCRIPTION
**Pull request by integraledelebesgue**
Closes [#2425](https://github.com/foundry-rs/starknet-foundry/issues/2425#issue-2495848646)
Closes [#2429](https://github.com/orgs/foundry-rs/projects/3?pane=issue&itemId=77954673)

## Introduced changes
- Cast commands committing transaction output properly formatted links
- Links differentiate Mainnet and Sepolia
- Functions used to prepare links have appropriate unit tests

## Checklist

- [x] Linked relevant issue
- [ ] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Comment by cptartur** [comment]
Please also link issue https://github.com/foundry-rs/starknet-foundry/issues/2425 to this PR

File: crates/sncast/src/response/explorer_link.rs
[the remark this answers is not in the document]
Comment (integraledelebesgue) [reply crates/sncast/src/response/explorer_link.rs:1]: Yes, assuming that Devnet has `chain_id = 'SN_SEPOLIA' or 'SN_MAINNET'` and the chosen explorer supports it (not every explorer shows Sepolia). Only those ID's are recognized as supported.

File: crates/sncast/src/helpers/block_explorer.rs [hunk 1]
```diff
@@ -36,82 +38,191 @@ pub trait LinkProvider {
...
+
+        let contract = links.next().unwrap();
+        let transaction = links.next().unwrap();
+
+        let (contract_response, transaction_response) =
+            tokio::join!(reqwest::get(contract), reqwest::get(transaction));
+
+        assert!(contract_response.is_ok());
+        assert!(transaction_response.is_ok());
+    }
```
Comment (cptartur) [inline crates/sncast/src/helpers/block_explorer.rs:192]: Not 100% like this being reliant on the order of links and a regex. But I see that rust doesn't really support lookbehind/lookahead in regex.

**Review by cptartur — APPROVED** [review]

File: crates/sncast/src/helpers/block_explorer.rs (hunk 1 above)
Comment (integraledelebesgue) [reply crates/sncast/src/helpers/block_explorer.rs:192]: Also, order of links is only cosmetic here. This function tests link format and no other logic

**Pull request merged**